pkg/clusterCache/ClusterCache.go

```diff
@@ -0,0 +1,256 @@
+package clusterCache
+
+import (
+	"errors"
+	"fmt"
+	clustercache "github.com/argoproj/gitops-engine/pkg/cache"
+	"github.com/argoproj/gitops-engine/pkg/utils/kube"
+	"github.com/caarlos0/env"
+	k8sUtils "github.com/devtron-labs/common-lib/utils/k8s"
+	"github.com/devtron-labs/common-lib/utils/k8s/health"
+	"github.com/devtron-labs/kubelink/bean"
+	repository "github.com/devtron-labs/kubelink/pkg/cluster"
+	"github.com/devtron-labs/kubelink/pkg/k8sInformer"
+	"github.com/devtron-labs/kubelink/pkg/util/argo"
+	"go.uber.org/zap"
+	"golang.org/x/sync/semaphore"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sync"
+)
+
+type ClusterCache interface {
+	//SyncClusterCache(clusterInfo bean.ClusterInfo) (clustercache.ClusterCache, error)
+	GetClusterCacheByClusterId(clusterId int) (clustercache.ClusterCache, error)
+}
+
+type ClusterCacheConfig struct {
+	ClusterIdList []int `env:"CLUSTER_ID_LIST" envSeparator:","`
+}
+
+func GetClusterCacheConfig() (*ClusterCacheConfig, error) {
+	cfg := &ClusterCacheConfig{}
+	err := env.Parse(cfg)
+	return cfg, err
+}
+
+type ClusterCacheImpl struct {
+	logger             *zap.SugaredLogger
+	clusterCacheConfig *ClusterCacheConfig
+	clusterRepository  repository.ClusterRepository
+	k8sUtil            *k8sUtils.K8sUtil
+	clustersCache      map[int]clustercache.ClusterCache
+	rwMutex            sync.RWMutex
+}
+
+func NewClusterCacheImpl(logger *zap.SugaredLogger, clusterCacheConfig *ClusterCacheConfig,
+	clusterRepository repository.ClusterRepository, k8sUtil *k8sUtils.K8sUtil) *ClusterCacheImpl {
+
+	clustersCache := make(map[int]clustercache.ClusterCache)
+	clusterCacheImpl := &ClusterCacheImpl{
+		logger:             logger,
+		clusterCacheConfig: clusterCacheConfig,
+		clusterRepository:  clusterRepository,
+		k8sUtil:            k8sUtil,
+		clustersCache:      clustersCache,
+	}
+
+	if len(clusterCacheConfig.ClusterIdList) > 0 {
+		err := clusterCacheImpl.SyncCache()
+		if err != nil {
+			return nil
+		}
+	}
+	return clusterCacheImpl
+}
+
+func (impl *ClusterCacheImpl) getClusterInfoByClusterId(clusterId int) (*bean.ClusterInfo, error) {
```

Comment:
need to check for updates in credentials of cluster just like doing in `k8sInformer`.